### DISCUSSION TITLE
How to replicate playground's auto list creation/insert/start feature?

### DISCUSSION THREAD

**Discussion by rte76702** [facebook/lexical#6092] [Q&A] [answered]
1. Currently in the [playground](https://playground.lexical.dev/) entering `- ` in a new line ie hyphen + space creates an unordered list, and entering `1. ` in a new line ie a number + period + space creates an ordered list.

      I imagined this would be possible by simply configuring the ListPlugin in some way but i can't seem to make it work.
      Using the example in the youtube video found in the [docs](https://lexical.dev/docs/getting-started/react): https://github.com/acywatson/glyf doesn't implement that feature.
      
      How does one go about implementing this? (Or a pointer to the playground implementation. Searched through the code, but can't seem to spot where the feature is implemented)

2. The playground only allows a list to be created in a new line, hence to create a nested list, tab would have to be used. As a low priority ask, how does one go about implementing (1) in a nested fashion, until the depth set in ListMaxIndentLevelPlugin?
I imagine the same justifications for the base case should also apply here, and there doesn't seem to be any html rule against interleaving `ul` and `ol`

**Comment by bakhshi** [comment, marked as answer]
@rte76702  The functionality you described in the first question is achieved via https://github.com/facebook/lexical/blob/5b8521c069c5e4a42546d2187a42070a09785ade/packages/lexical-playground/src/Editor.tsx#L178

Similar behavior for headings can be observed if we type #, ##, or ### followed by a space. H1, H2, or H3 headings will be inserted correspondingly.

**Reply by SaintPepsi** [reply]
This works, but there seem to be strange conditions under which it's not automatically creating the desired elements, for example in our application we have 2 different pages which render basically the exact same content minus a couple fluffy elements for UX. the Rich Text Editor we build with lexical is exactly the same, context is the same, everything is the same. but on 1 page automatic lists don't work. which makes no sense.
